src/Sprites/ArtWriter.cpp

```diff
@@ -86,6 +87,37 @@ namespace ArtWriter
 
 	void WriteAnimation(Stream::SeekableWriter& seekableWriter, const Animation& animation)
 	{
+		seekableWriter.Write(animation.unknown);
+		seekableWriter.Write(animation.selectionRect);
+		seekableWriter.Write(animation.pixelDisplacement);
+		seekableWriter.Write(animation.unknown2);
+		
+		std::size_t frameCount = animation.frames.size();
+		if (frameCount > UINT32_MAX) {
+			throw std::runtime_error("There are too many frames in animation to write");
+		}
+		seekableWriter.Write(frameCount);
```

Comment:
Here `frameCount` is a `std::size_t`. I think it needs a `static_cast<uint32_t>`.

Reply:
Nice catch, I'll fix it.